### ISSUE TITLE
Remove Shortcuts Button

### ISSUE THREAD

**Issue by colintheshots** [mozilla-mobile/fenix#5567]
### Why/User Benefit/User Problem
The Shortcuts button now does nothing, since it's the default UX behavior for the search screen. It's always activated.

Some users find this annoying: https://www.reddit.com/r/firefox/comments/d7ak8s/does_anyone_know_how_can_i_disable_the_shortcuts/

### What/Requirements
Remove the button.

### Acceptance Criteria (how do I know when I’m done?)
The button will no longer appear on the search screen.

**Comment by nisrulz** [comment]
Hi,

I'd like to take a stab at this.

**Comment by ekager** [comment]
This still needs UX feedback

**Comment by sblatz** [comment]
@colintheshots Please do not add feature requests to the sprint board until they have been prioritized by @vesta0.

**Comment by sblatz** [comment]
It's not true that this button "does nothing." I agree we should consider changing the default to not show search suggestions, but pressing it *does* hide them.

**Comment by brampitoyo** [comment]
@colintheshots Thanks for filing this issue!

* Let’s remove the Shortcuts button
* Let’s add a new item under Settings → Search called “Show search shortcuts”, that allows users to turn off search shortcuts. It defaults to ON, and works exactly as @sblatz has described

Thanks heaps for the idea, @sblatz.

**Comment by Cheap-Skate** [comment]
Now the Shortcuts button is removed could the Scan button be removed or placed elsewhere? 
That way we'd all win back a bit of vertical space to fill with search results. The small number of search results is IMHO a big step back from Fennec, see eg #1161, #479.

eg
- put it where it is in Fennec, on the right hand side of the URL bar?
or
- leave it where it is, but make it disappear as soon as I start typing in the URL bar and give the vertical space back to search results (I'm unlikely to want to Scan after I have started typing???)

**Comment by brampitoyo** [comment]
@Cheap-Skate File an issue, and I’ll help triage it! When our team has some time available, we’ll consider it :-)

**Comment by sv-ohorvath** [comment]
Verified the shortcuts button is hidden now, the new preference in Search settings and no new issues found. 
build: Nightly 10/4
device: Pixel 3 (Android 9), Huawei P9 lite (Android 6).

**Comment by cadeyrn** [comment]
Now it's no longer possible to change the search engine after you entered something into the address bar. This can be very annoying if you notice that you selected the wrong search engine _after_ entering the search term because you have to remove the whole input, change the seach engine and type it again… It's also a functional regression for the Fennec -> Fenix migration where you _can_ change the search engine after entering the search term.

If the only reason to remove the button was that some people find it annoying, is the new preference in the search engines to disable the shortcuts not sufficient?

/cc @colintheshots @brampitoyo

**Comment by klint** [comment]
Maybe the button could be back when the setting is Off?

**Comment by Cheap-Skate** [comment]
Question:- How will the newly adopted search behavior (show search engines before typing, search engines disappear when typing starts) work with Top Sites which IIUC is coming soon? Presumably Top Sites will appear before I type & use a lot of the screen real estate which  shows search engines. ie won't Top Sites and search engine list conflict over the same screen real estate?

Suggestion:- How about select a search engine at any time by tapping the search favicon at the left end of the URL bar (Google for me).

Agree current behavior is not as good as Fennec - again IMHO showing how excellent the Fennec Awesomescreen is/was. I shall really miss it.

**Comment by Cheap-Skate** [comment]
Re tapping the search favicon, Samsung Internet does this, I like it. Down pointing caret next to the search favicon in URL bar, tapping the search favicon drops down a list of search engines.
![Screenshot_20191006-175224_Samsung Internet Beta](https://user-images.githubusercontent.com/40568887/66272478-3d09c400-e862-11e9-8d91-6a798b400487.jpg)
![Screenshot_20191006-175248_Samsung Internet Beta](https://user-images.githubusercontent.com/40568887/66272479-4135e180-e862-11e9-84e6-e74652cfd17f.jpg)

**Comment by yoasif** [comment]
@cadeyrn Can you open a new issue for this regression? There is also a post on reddit: https://www.reddit.com/r/firefox/comments/de03eu/why_search_with_was_removed_now_u_can_change/

**Comment by brampitoyo** [comment]
@cadeyrn wrote:

> If the only reason to remove the button was that some people find it annoying, is the new preference in the search engines to disable the shortcuts not sufficient?

I agree with you that having this preference is totally sufficient.

I wrote in https://github.com/mozilla-mobile/fenix/issues/5567#issuecomment-537318950:

>Let’s add a new item under Settings → Search called “Show search shortcuts”, that allows users to turn off search shortcuts. **It defaults to ON** […]

The default value should be ON. Meaning: the “Shortcuts” button will be available to all users unless you manually turn it off.

Would this fix the issue?

**Comment by klint** [comment]
> The default value should be ON. Meaning: the “Shortcuts” button will be available to all users unless you manually turn it off.
Sorry, I'm lost. Are you talking about the "Shortcuts" button or the shortcut entries above with the name of each search engine??? In the latest nightly, the setting deals with the shortcut entries, and the "Shortcuts" button is always hidden. What is the new fix you are suggesting then? Thanks :)

**Comment by klint** [comment]
> The default value should be ON. Meaning: the “Shortcuts” button will be available to all users unless you manually turn it off.

Sorry, I'm lost: are you talking about the "Shortcuts" button or the shortcut entries above with the name of each search engine??? 
In the latest nightly, the setting deals with the shortcut entries, and the "Shortcuts" button is always hidden. What is the new fix you are suggesting then? 
Thanks :)

**Comment by Cheap-Skate** [comment]
Sorry for confusion, let me try again. My suggestion is to bring back the "Shortcuts" button which used to show/hide the list of search shortcuts, but instead of having it as a distinct button at the bottom of the page, put its behavior into the "currently selected" search favicon at the left hand side of the text entry bar. i.e. tap the "currently selected" search favicon to bring up the list of search shortcuts. This is how Samsung Internet does it.

This fixes cadeyrn's issue:-

> Now it's no longer possible to change the search engine after you entered something into the address bar. ... It's also a functional regression for the Fennec -> Fenix migration where you can change the search engine after entering the search term.

It might also fix a future problem where Top Sites will conflict for screen real estate with current default behavior of showing the list of search shortcuts on opening a new tab:-

Top Sites are coming. Presumably the default behavior of Top Sites will be to appear as soon as I open a new tab with no text in the search bar. But Fenix's current default behavior is to show a list of search shortcuts when I open a new tab. What should Fenix show on opening a new tab:- Top Sites or list of search shortcuts? My proposed solution fixes this potential issue as well.

**Comment by klint** [comment]
Thanks @Cheap-Skate for that clarification.
@brampitoyo, is it the same fix you are referring to? :)

**Comment by sblatz** [comment]
> Now it's no longer possible to change the search engine after you entered something into the address bar. ...

If you being a search and then backspace until the text is empty, you will be presented with the search engine switcher again, so you can change it there. That being said, that's not necessarily super discoverable or intuitive.

**Comment by brampitoyo** [comment]
@Cheap-Skate Thanks for clarifying your suggestions on https://github.com/mozilla-mobile/fenix/issues/5567#issuecomment-539019834.

> […] instead of having [search shortcuts] as a distinct button at the bottom of the page […] tap the "currently selected" search favicon to bring up the list of search shortcuts.
> 
> This fixes @cadeyrn’s issue: […] no longer possible to change the search engine after you entered something into the address bar.
> 
> It might also fix a future problem where Top Sites will conflict for screen real estate with […] the list of search shortcuts.

This is a totally separate issue from removing the Shortcuts button. So, will you file a new issue for us? We will definitely consider this as part of the larger search experience design (#5457).

Thanks heaps!
